Correctly set gtest 1.8 root directory

dgossow opened the pull request:
In gtest version 1.8 or higher, there is a CMakeLists.txt in the
parent directory of the gtest source directory which we need to find.
This can be either /usr/src/googletest or ${CMAKE_SOURCE_DIR}/googletest.

The current version would only check in /usr/src/googletest even if
gtest was found in the project's src/ directory.

This would break the build on Ubuntu 18.04, where the system install of gmock
is version 1.8, if your project includes a newer version of gtest in its src/ directory.
The result would be that headers from src/ would be used, but the test
executables would be linked to the binaries from /usr/src.

dgossow commented:
@wjwwood @dirk-thomas Could you take a look at this?

dgossow commented:
The CI build fails like this:

```
Unable to download 3.4_with_system_site_packages archive. The archive may not exist. Please consider a different version.
```

Does not seem related though.

I still need to test this on Ubuntu 16.04 and 18.04 with and without gtest sources inside the workspace to make sure it works.

dirk-thomas commented:
> The CI build fails like this:

@dgossow Please rebase your change on top of the latest state of the target branch.

dirk-thomas commented:
@dgossow Friendly ping.

dgossow committed:
Correctly set gtest 1.8 root directory

In gtest version 1.8 or higher, there is a CMakeLists.txt in the
parent directory of the gtest source directory which we need to find.
This can be either /usr/src/googletest or
${CMAKE_SOURCE_DIR}/googletest.

The current version would only check in /usr/src/googletest even if
gtest was found in the project's src/ directory.

This would break the build on Ubuntu 18.04, where the system install of
gmock is version 1.8, if your project includes a newer version of gtest
in its src/ directory. The result would be that headers from src/ would
be used, but the test executables would be linked to the binaries from
/usr/src.

dgossow commented:
Sorry, I was distracted. So I essentially rebased this on top of kinetic-devel like requested.

I also tested this on 16.04 and 18.04.

16.04: 
gtest found under /usr/src/gmock
gmock found under /usr/src/gmock

18.04:
gtest found under /usr/src/googletest
gmock found under /usr/src/googletest

Finding the current version (master) of gmock/gtest under src/googletest works in both cases.

dgossow commented:
@dirk-thomas comments addressed

dirk-thomas commented:
Thanks for the patch.
